addons/stock_dropshipping/models/stock.py

```diff
@@ -21,6 +21,16 @@ def _get_partner_id(self, values, rule):
             return False
         return super()._get_partner_id(values, rule)
 
+    @api.depends('action')
+    def _compute_picking_type_code_domain(self):
+        remaining = self.browse()
+        for rule in self:
+            if rule.action == 'buy':
+                rule.picking_type_code_domain = ['incoming', 'dropship']
```

Comment:
I would be explicit and build the list across module. So starting from empty list and adding code. It will be more maintainable and more extensible. Thank you